frontend/app/utils/ReferralData.scala

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import play.api.mvc.{Cookie, RequestHeader}
+
+case class ReferralData(campaignCode: Option[String], url: Option[String], pageviewId: Option[String])
+
+object ReferralData {
+
+  val CampaignCodeKey = "mem_campaign_code"
+  val UrlKey = "gu_mem_ref_url"
+  val PageviewIdKey = "gu_refpvid"
+
+  def makeCookies(implicit request: RequestHeader): Seq[Cookie] = {
+    val refUrl = request.headers.get("referer").map(Cookie(ReferralData.UrlKey, _))
```

Comment:
also according to https://github.com/guardian/membership-frontend/blob/master/frontend/app/controllers/Info.scala#L93 this is a Cached [for 60 seconds](https://github.com/guardian/membership-frontend/blob/master/frontend/app/controllers/Cached.scala#L15) method, and we can't set a cookie based on the referrer in a cached method